### PR TITLE
Issue #12678: Re-enable action lint

### LINKED ISSUE #12678: Re-enable action lint
Actionlint docker image does not like arguments to the container, see https://github.com/checkstyle/checkstyle/actions/runs/3998588410/jobs/6861429747:

```
could not read "deprecated.*'": open deprecated.*': no such file or directory


```

It is splitting the arguments even if we use `=` or enclose them in quotes. Let's disable actionlint for now to unblock restoration of diff report generation with patch config, and deal with actionlint later.

### PR DESCRIPTION
**Pull request by stoyanK7**
Resolves #12678

https://github.com/checkstyle/checkstyle/issues/12678#issuecomment-1488545090

---
The following errors were present:
https://github.com/stoyanK7/checkstyle/actions/runs/4553573127/jobs/8030294702#step:4:6
```
.github/workflows/release-new-milestone-and-issues-in-other-repos.yml:27:92: property "concurrency-group" is not defined in object type {version: any} [expression]
.github/workflows/release-publish-releasenotes-twitter.yml:23:92: property "concurrency-group" is not defined in object type {version: any} [expression]
.github/workflows/release-update-github-io.yml:23:92: property "concurrency-group" is not defined in object type {version: any} [expression]
.github/workflows/release-update-github-page.yml:23:92: property "concurrency-group" is not defined in object type {version: any} [expression]
.github/workflows/release-upload-all-jar.yml:26:92: property "concurrency-group" is not defined in object type {version: any} [expression]
```
Success run after they were fixed: https://github.com/stoyanK7/checkstyle/actions/runs/4566213905/jobs/8058397704#step:4:5

### CODE REVIEW

**Comment by stoyanK7** [comment]
Aand there's 2 more errors to fix..

**Comment by stoyanK7** [comment]
https://github.com/checkstyle/checkstyle/actions/runs/4568150162/jobs/8062829735?pr=12938#step:4:6
```
.github/workflows/site.yml:120:9: shellcheck reported issue in this script: SC2129:style:2:1: Consider using { cmd1; cmd2; } >> file instead of individual redirects [shellcheck]
    |
120 |         run: |
    |         ^~~~
.github/workflows/site.yml:153:9: shellcheck reported issue in this script: SC2129:style:2:1: Consider using { cmd1; cmd2; } >> file instead of individual redirects [shellcheck]
    |
153 |         run: |
    |         ^~~~
```

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: .github/workflows/actionlint.yml [hunk 1]
```diff
@@ -14,8 +18,10 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
       - name: Check workflow files
         uses: docker://rhysd/actionlint:latest
         with:
-          args: -color
+          args: -ignore ".* was deprecated.*"
```
Comment (nrmancuso) [inline .github/workflows/actionlint.yml:27]: Please open an issue to resolve these if we don’t already have one, and leave until comment above this suppression

**Review by nrmancuso — CHANGES_REQUESTED** [review]

File: .github/workflows/actionlint.yml (hunk 1 above)
Comment (stoyanK7) [reply .github/workflows/actionlint.yml:28]: Done. There is an issue at #12696 already.

**Review by nrmancuso — APPROVED** [review]
Awesome @stoyanK7, thanks a lot for taking care of this. Ok to merge as CI passes.

**Pull request merged**